clients/java/src/main/java/marquez/client/MarquezWriteOnlyClientImpl.java

```diff
@@ -89,7 +88,7 @@ public void createRun(String namespaceName, String jobName, RunMeta runMeta) {
   public void markRunAs(String runId, RunState runState, Instant at) {
     Map<String, Object> queryParams =
         at == null ? null : ImmutableMap.of("at", ISO_INSTANT.format(at));
-    backend.post(path(runTransitionPath(runId, runState), queryParams));
+    backend.post(queryPath(runTransitionPath(runId, runState), queryParams), null);
```

Comment:
Can we keep the method name as `path()`? or `pathWithQueryParams()`? Using `queryPath()` reads more like the query params for the path.

Reply:
What about `attachQueryParams`? I think it's more accurate about what this function does.

Reply:
Done